readme.md
Top 100 Brands

changelog.md
# Changelog
2014-05-23 	Project Start, Setup, Map Design
2014-05-24	Basis Styles & Functions, Mapdata
2014-05-25	Basis Styles & Functions, Mapdata
2014-05-26	Country Layer, Filter, Skrollr
2014-05-27	Navigation, Transitions, Tooltip
2014-05-28	Logos
2014-05-29	Info
2014-05-31	Navigation Jump, Mobile compatibility, Touch compatibility, Firefox compatibility
2014-06-01	Colors, Social Sharing
2014-06-13	Marker Collision, Page Loading, A Logo
2014-06-14	Design Tweaks
2014-06-15	Design Tweaks, Hash Navigation
2014-06-16	Navigation Jump Filter Fix, Marker Collision, Design Tweaks
2014-06-20	2013, 2012
2014-06-23	2011, 2010
2014-06-24	More Logos, Brand Charts
2014-06-25	Brand Charts, Firefox tweaks
2014-06-26	Design Tweaks
2014-06-27	Design Tweaks, Tracking, Touch tweaks
2014-06-30	Touch tweaks, Ancient browser info, Hash navigation scroll, Mobile tweaks
2014-07-01	IE11 compatibility
2014-08-03	Local Map tiles, Marker & Popup tweaks
2014-08-08	Map design update
2014-08-10	Design tweaks (Marker)
2014-08-11	Design tweaks (Marker, Popup, Map)
2014-08-12	Bugfixing (Hash Navigation), Design tweaks, Logo graphics attribution
2014-08-17	Design tweaks (Popup)
2014-08-18	Design tweaks (Popup)
2014-08-21	Sources
2015-04-05	Design tweaks
2015-06-03	2015 data update, design tweaks, retina charts
2018-10-26	Data updated for 2016, 2017 and 2018, small design tweeks/fixes

# Planned Features
- 	Brand prominence
- 	Brand comparison
- 	Top brands per country
- 	Top brands per category as table/chart

develop/js/pace.js
/*	Patched by Alexander Johmann
		+ document.querySelector('h1.desktop span span').setAttribute("data-progress-text", "" + (0 | this.progress))
		+ document.querySelector('h1.tablet span span').setAttribute("data-progress-text", "" + (0 | this.progress))
		+ document.querySelector('h1.phone span span').setAttribute("data-progress-text", "" + (0 | this.progress))
		+ paceOptions var
*/

paceOptions = {
	restartOnPushState: false
};

/*! pace 0.5.4 */ (function() {
		var a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, u, v, w, x, y, z, A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W = [].slice,
				X = {}.hasOwnProperty,
				Y = function(a, b) {
						function c() {
								this.constructor = a
						}
						for (var d in b) X.call(b, d) && (a[d] = b[d]);
						return c.prototype = b.prototype, a.prototype = new c, a.__super__ = b.prototype, a
				},
				Z = [].indexOf ||
		function(a) {
				for (var b = 0, c = this.length; c > b; b++) if (b in this && this[b] === a) return b;
				return -1
		};
		for (t = {
				catchupTime: 500,
				initialRate: .03,
				minTime: 500,
				ghostTime: 500,
				maxProgressPerFrame: 10,
				easeFactor: 1.25,
				startOnPageLoad: !0,
				restartOnPushState: !0,
				restartOnRequestAfter: 500,
				target: "body",
				elements: {
						checkInterval: 100,
						selectors: ["body"]
				},
				eventLag: {
						minSamples: 10,
						sampleCount: 3,
						lagThreshold: 3
				},
				ajax: {
						trackMethods: ["GET"],
						trackWebSockets: !0,
						ignoreURLs: []
				}
		}, B = function() {
				var a;
				return null != (a = "undefined" != typeof performance && null !== performance && "function" == typeof performance.now ? performance.now() : void 0) ? a : +new Date
		}, D = window.requestAnimationFrame || window.mozRequestAnimationFrame || window.webkitRequestAnimationFrame || window.msRequestAnimationFrame, s = window.cancelAnimationFrame || window.mozCancelAnimationFrame, null == D && (D = function(a) {
				return setTimeout(a, 50)
		}, s = function(a) {
				return clearTimeout(a)
		}), F = function(a) {
				var b, c;
				return b = B(), (c = function() {
						var d;
						return d = B() - b, d >= 33 ? (b = B(), a(d, function() {
								return D(c)
						})) : setTimeout(c, 33 - d)
				})()
		}, E = function() {
				var a, b, c;
				return c = arguments[0], b = arguments[1], a = 3 <= arguments.length ? W.call(arguments, 2) : [], "function" == typeof c[b] ? c[b].apply(c, a) : c[b]
		}, u = function() {
				var a, b, c, d, e, f, g;
				for (b = arguments[0], d = 2 <= arguments.length ? W.call(arguments, 1) : [], f = 0, g = d.length; g > f; f++) if (c = d[f]) for (a in c) X.call(c, a) && (e = c[a], null != b[a] && "object" == typeof b[a] && null != e && "object" == typeof e ? u(b[a], e) : b[a] = e);
				return b
		}, p = function(a) {
				var b, c, d, e, f;
				for (c = b = 0, e = 0, f = a.length; f > e; e++) d = a[e], c += Math.abs(d), b++;
				return c / b
		}, w = function(a, b) {
				var c, d, e;
				if (null == a && (a = "options"), null == b && (b = !0), e = document.querySelector("[data-pace-" + a + "]")) {
						if (c = e.getAttribute("data-pace-" + a), !b) return c;
						try {
								return JSON.parse(c)
						} catch (f) {
								return d = f, "undefined" != typeof console && null !== console ? console.error("Error parsing inline pace options", d) : void 0
						}
				}
		}, g = function() {
				function a() {}
				return a.prototype.on = function(a, b, c, d) {
						var e;
						return null == d && (d = !1), null == this.bindings && (this.bindings = {}), null == (e = this.bindings)[a] && (e[a] = []), this.bindings[a].push({
								handler: b,
								ctx: c,
								once: d
						})
				}, a.prototype.once = function(a, b, c) {
						return this.on(a, b, c, !0)
				}, a.prototype.off = function(a, b) {
						var c, d, e;
						if (null != (null != (d = this.bindings) ? d[a] : void 0)) {
								if (null == b) return delete this.bindings[a];
								for (c = 0, e = []; c < this.bindings[a].length;) e.push(this.bindings[a][c].handler === b ? this.bindings[a].splice(c, 1) : c++);
								return e
						}
				}, a.prototype.trigger = function() {
						var a, b, c, d, e, f, g, h, i;
						if (c = arguments[0], a = 2 <= arguments.length ? W.call(arguments, 1) : [], null != (g = this.bindings) ? g[c] : void 0) {
								for (e = 0, i = []; e < this.bindings[c].length;) h = this.bindings[c][e], d = h.handler, b = h.ctx, f = h.once, d.apply(null != b ? b : this, a), i.push(f ? this.bindings[c].splice(e, 1) : e++);
								return i
						}
				}, a
		}(), null == window.Pace && (window.Pace = {}), u(Pace, g.prototype), C = Pace.options = u({}, t, window.paceOptions, w()), T = ["ajax", "document", "eventLag", "elements"], P = 0, R = T.length; R > P; P++) J = T[P], C[J] === !0 && (C[J] = t[J]);
		i = function(a) {
				function b() {
						return U = b.__super__.constructor.apply(this, arguments)
				}
				return Y(b, a), b
		}(Error), b = function() {
				function a() {
						this.progress = 0
				}
				return a.prototype.getElement = function() {
						var a;
						if (null == this.el) {
								if (a = document.querySelector(C.target), !a) throw new i;
								this.el = document.createElement("div"), this.el.className = "pace pace-active", document.body.className = document.body.className.replace(/pace-done/g, ""), document.body.className += " pace-running", this.el.innerHTML = '<div class="pace-progress">n	<div class="pace-progress-inner"></div>n</div>n<div class="pace-activity"></div>', null != a.firstChild ? a.insertBefore(this.el, a.firstChild) : a.appendChild(this.el)
						}
						return this.el
				}, a.prototype.finish = function() {
						var a;
						return a = this.getElement(), a.className = a.className.replace("pace-active", ""), a.className += " pace-inactive", document.body.className = document.body.className.replace("pace-running", ""), document.body.className += " pace-done"
				}, a.prototype.update = function(a) {
						return this.progress = a, this.render()
				}, a.prototype.destroy = function() {
						try {
								this.getElement().parentNode.removeChild(this.getElement())
						} catch (a) {
								i = a
						}
						return this.el = void 0
				}, a.prototype.render = function() {
						var a, b;
						return null == document.querySelector(C.target) ? !1 : (a = this.getElement(), a.children[0].style.width = "" + this.progress + "%", (!this.lastRenderedProgress || this.lastRenderedProgress | 0 !== this.progress | 0) && (a.children[0].setAttribute("data-progress-text", "" + (0 | this.progress) + "%"), document.querySelector('h1.desktop span span').setAttribute("data-progress-text", "" + (0 | this.progress)), document.querySelector('h1.tablet span span').setAttribute("data-progress-text", "" + (0 | this.progress)), document.querySelector('h1.phone span span').setAttribute("data-progress-text", "" + (0 | this.progress)), this.progress >= 100 ? b = "99" : (b = this.progress < 10 ? "0" : "", b += 0 | this.progress), a.children[0].setAttribute("data-progress", "" + b)), this.lastRenderedProgress = this.progress)
				}, a.prototype.done = function() {
						return this.progress >= 100
				}, a
		}(), h = function() {
				function a() {
						this.bindings = {}
				}
				return a.prototype.trigger = function(a, b) {
						var c, d, e, f, g;
						if (null != this.bindings[a]) {
								for (f = this.bindings[a], g = [], d = 0, e = f.length; e > d; d++) c = f[d], g.push(c.call(this, b));
								return g
						}
				}, a.prototype.on = function(a, b) {
						var c;
						return null == (c = this.bindings)[a] && (c[a] = []), this.bindings[a].push(b)
				}, a
		}(), O = window.XMLHttpRequest, N = window.XDomainRequest, M = window.WebSocket, v = function(a, b) {
				var c, d, e, f;
				f = [];
				for (d in b.prototype) try {
						e = b.prototype[d], f.push(null == a[d] && "function" != typeof e ? a[d] = e : void 0)
				} catch (g) {
						c = g
				}
				return f
		}, z = [], Pace.ignore = function() {
				var a, b, c;
				return b = arguments[0], a = 2 <= arguments.length ? W.call(arguments, 1) : [], z.unshift("ignore"), c = b.apply(null, a), z.shift(), c
		}, Pace.track = function() {
				var a, b, c;
				return b = arguments[0], a = 2 <= arguments.length ? W.call(arguments, 1) : [], z.unshift("track"), c = b.apply(null, a), z.shift(), c
		}, I = function(a) {
				var b;
				if (null == a && (a = "GET"), "track" === z[0]) return "force";
				if (!z.length && C.ajax) {
						if ("socket" === a && C.ajax.trackWebSockets) return !0;
						if (b = a.toUpperCase(), Z.call(C.ajax.trackMethods, b) >= 0) return !0
				}
				return !1
		}, j = function(a) {
				function b() {
						var a, c = this;
						b.__super__.constructor.apply(this, arguments), a = function(a) {
								var b;
								return b = a.open, a.open = function(d, e) {
										return I(d) && c.trigger("request", {
												type: d,
												url: e,
												request: a
										}), b.apply(a, arguments)
								}
						}, window.XMLHttpRequest = function(b) {
								var c;
								return c = new O(b), a(c), c
						}, v(window.XMLHttpRequest, O), null != N && (window.XDomainRequest = function() {
								var b;
								return b = new N, a(b), b
						}, v(window.XDomainRequest, N)), null != M && C.ajax.trackWebSockets && (window.WebSocket = function(a, b) {
								var d;
								return d = null != b ? new M(a, b) : new M(a), I("socket") && c.trigger("request", {
										type: "socket",
										url: a,
										protocols: b,
										request: d
								}), d
						}, v(window.WebSocket, M))
				}
				return Y(b, a), b
		}(h), Q = null, x = function() {
				return null == Q && (Q = new j), Q
		}, H = function(a) {
				var b, c, d, e;
				for (e = C.ajax.ignoreURLs, c = 0, d = e.length; d > c; c++) if (b = e[c], "string" == typeof b) {
						if (-1 !== a.indexOf(b)) return !0
				} else if (b.test(a)) return !0;
				return !1
		}, x().on("request", function(b) {
				var c, d, e, f, g;
				return f = b.type, e = b.request, g = b.url, H(g) ? void 0 : Pace.running || C.restartOnRequestAfter === !1 && "force" !== I(f) ? void 0 : (d = arguments, c = C.restartOnRequestAfter || 0, "boolean" == typeof c && (c = 0), setTimeout(function() {
						var b, c, g, h, i, j;
						if (b = "socket" === f ? e.readyState < 2 : 0 < (h = e.readyState) && 4 > h) {
								for (Pace.restart(), i = Pace.sources, j = [], c = 0, g = i.length; g > c; c++) {
										if (J = i[c], J instanceof a) {
												J.watch.apply(J, d);
												break
										}
										j.push(void 0)
								}
								return j
						}
				}, c))
		}), a = function() {
				function a() {
						var a = this;
						this.elements = [], x().on("request", function() {
								return a.watch.apply(a, arguments)
						})
				}
				return a.prototype.watch = function(a) {
						var b, c, d, e;
						return d = a.type, b = a.request, e = a.url, H(e) ? void 0 : (c = "socket" === d ? new m(b) : new n(b), this.elements.push(c))
				}, a
		}(), n = function() {
				function a(a) {
						var b, c, d, e, f, g, h = this;
						if (this.progress = 0, null != window.ProgressEvent) for (c = null, a.addEventListener("progress", function(a) {
								return h.progress = a.lengthComputable ? 100 * a.loaded / a.total : h.progress + (100 - h.progress) / 2
						}), g = ["load", "abort", "timeout", "error"], d = 0, e = g.length; e > d; d++) b = g[d], a.addEventListener(b, function() {
								return h.progress = 100
						});
						else f = a.onreadystatechange, a.onreadystatechange = function() {
								var b;
								return 0 === (b = a.readyState) || 4 === b ? h.progress = 100 : 3 === a.readyState && (h.progress = 50), "function" == typeof f ? f.apply(null, arguments) : void 0
						}
				}
				return a
		}(), m = function() {
				function a(a) {
						var b, c, d, e, f = this;
						for (this.progress = 0, e = ["error", "open"], c = 0, d = e.length; d > c; c++) b = e[c], a.addEventListener(b, function() {
								return f.progress = 100
						})
				}
				return a
		}(), d = function() {
				function a(a) {
						var b, c, d, f;
						for (null == a && (a = {}), this.elements = [], null == a.selectors && (a.selectors = []), f = a.selectors, c = 0, d = f.length; d > c; c++) b = f[c], this.elements.push(new e(b))
				}
				return a
		}(), e = function() {
				function a(a) {
						this.selector = a, this.progress = 0, this.check()
				}
				return a.prototype.check = function() {
						var a = this;
						return document.querySelector(this.selector) ? this.done() : setTimeout(function() {
								return a.check()
						}, C.elements.checkInterval)
				}, a.prototype.done = function() {
						return this.progress = 100
				}, a
		}(), c = function() {
				function a() {
						var a, b, c = this;
						this.progress = null != (b = this.states[document.readyState]) ? b : 100, a = document.onreadystatechange, document.onreadystatechange = function() {
								return null != c.states[document.readyState] && (c.progress = c.states[document.readyState]), "function" == typeof a ? a.apply(null, arguments) : void 0
						}
				}
				return a.prototype.states = {
						loading: 0,
						interactive: 50,
						complete: 100
				}, a
		}(), f = function() {
				function a() {
						var a, b, c, d, e, f = this;
						this.progress = 0, a = 0, e = [], d = 0, c = B(), b = setInterval(function() {
								var g;
								return g = B() - c - 50, c = B(), e.push(g), e.length > C.eventLag.sampleCount && e.shift(), a = p(e), ++d >= C.eventLag.minSamples && a < C.eventLag.lagThreshold ? (f.progress = 100, clearInterval(b)) : f.progress = 100 * (3 / (a + 3))
						}, 50)
				}
				return a
		}(), l = function() {
				function a(a) {
						this.source = a, this.last = this.sinceLastUpdate = 0, this.rate = C.initialRate, this.catchup = 0, this.progress = this.lastProgress = 0, null != this.source && (this.progress = E(this.source, "progress"))
				}
				return a.prototype.tick = function(a, b) {
						var c;
						return null == b && (b = E(this.source, "progress")), b >= 100 && (this.done = !0), b === this.last ? this.sinceLastUpdate += a : (this.sinceLastUpdate && (this.rate = (b - this.last) / this.sinceLastUpdate), this.catchup = (b - this.progress) / C.catchupTime, this.sinceLastUpdate = 0, this.last = b), b > this.progress && (this.progress += this.catchup * a), c = 1 - Math.pow(this.progress / 100, C.easeFactor), this.progress += c * this.rate * a, this.progress = Math.min(this.lastProgress + C.maxProgressPerFrame, this.progress), this.progress = Math.max(0, this.progress), this.progress = Math.min(100, this.progress), this.lastProgress = this.progress, this.progress
				}, a
		}(), K = null, G = null, q = null, L = null, o = null, r = null, Pace.running = !1, y = function() {
				return C.restartOnPushState ? Pace.restart() : void 0
		}, null != window.history.pushState && (S = window.history.pushState, window.history.pushState = function() {
				return y(), S.apply(window.history, arguments)
		}), null != window.history.replaceState && (V = window.history.replaceState, window.history.replaceState = function() {
				return y(), V.apply(window.history, arguments)
		}), k = {
				ajax: a,
				elements: d,
				document: c,
				eventLag: f
		}, (A = function() {
				var a, c, d, e, f, g, h, i;
				for (Pace.sources = K = [], g = ["ajax", "elements", "document", "eventLag"], c = 0, e = g.length; e > c; c++) a = g[c], C[a] !== !1 && K.push(new k[a](C[a]));
				for (i = null != (h = C.extraSources) ? h : [], d = 0, f = i.length; f > d; d++) J = i[d], K.push(new J(C));
				return Pace.bar = q = new b, G = [], L = new l
		})(), Pace.stop = function() {
				return Pace.trigger("stop"), Pace.running = !1, q.destroy(), r = !0, null != o && ("function" == typeof s && s(o), o = null), A()
		}, Pace.restart = function() {
				return Pace.trigger("restart"), Pace.stop(), Pace.start()
		}, Pace.go = function() {
				var a;
				return Pace.running = !0, q.render(), a = B(), r = !1, o = F(function(b, c) {
						var d, e, f, g, h, i, j, k, m, n, o, p, s, t, u, v;
						for (k = 100 - q.progress, e = o = 0, f = !0, i = p = 0, t = K.length; t > p; i = ++p) for (J = K[i], n = null != G[i] ? G[i] : G[i] = [], h = null != (v = J.elements) ? v : [J], j = s = 0, u = h.length; u > s; j = ++s) g = h[j], m = null != n[j] ? n[j] : n[j] = new l(g), f &= m.done, m.done || (e++, o += m.tick(b));
						return d = o / e, q.update(L.tick(b, d)), q.done() || f || r ? (q.update(100), Pace.trigger("done"), setTimeout(function() {
								return q.finish(), Pace.running = !1, Pace.trigger("hide")
						}, Math.max(C.ghostTime, Math.max(C.minTime - (B() - a), 0)))) : c()
				})
		}, Pace.start = function(a) {
				u(C, a), Pace.running = !0;
				try {
						q.render()
				} catch (b) {
						i = b
				}
				return document.querySelector(".pace") ? (Pace.trigger("start"), Pace.go()) : setTimeout(Pace.start, 50)
		}, "function" == typeof define && define.amd ? define(function() {
				return Pace
		}) : "object" == typeof exports ? module.exports = Pace : C.startOnPageLoad && Pace.start()
}).call(this);

develop/js/map.js
/******************************************
* TOP 100 BRANDS                          *
* Alexander Johmann, 2014-2018            *
******************************************/

var map, dataLayer, countryLayer = [], fadeout = false;

var colors = {
	technology: '#3eb9e3',
	technologyRGB: '62, 185, 227',
	telecoms: '#ff8a54',
	telecomsRGB: '255, 138, 84', 
	retail: '#a7da4d',
	retailRGB : '167, 218, 77',
	fastfood: '#ef5b63',
	fastfoodRGB: '239, 91, 99',
	softdrinks: '#63a7af',
	softdrinksRGB: '99, 167, 175',
	cars: '#ec77dc',
	carsRGB: '236, 119, 220',
	globalbanks: '#a992ed',
	globalbanksRGB: '169, 146, 237',
	luxury: '#ffec7d',
	luxuryRGB: '255, 236, 125',
	oilandgas: '#d9a162',
	oilandgasRGB: '217, 161, 98'
};

var charJSPersonalDefaultOptionsHorizontalBar = {
	barStrokeWidth: 0,
	barValueSpacing: 0,
	inGraphDataShow: true,
	xAxisBottom: false,
	inGraphDataFontColor: '#000',
	inGraphDataXPosition: 3,
	inGraphDataPaddingX: -7,
	inGraphDataAlign: 'right',
	inGraphDataTmpl: 'M$ <%=v3%>',
	inGraphDataFontFamily: "proxima-nova",
	inGraphDataFontSize: 12,
	scaleFontColor: '#666',
	scaleFontFamily: 'inconsolata',
	scaleFontSize: 14,
	barDatasetSpacing: 1,
	responsive: true,
	graphMin: 0,
	scaleShowGridLines: false,
	scaleShowLine: false,
	showYAxisMin: false,
	animation: true,
	yAxisRight: false,
	maintainAspectRatio: false,
};

function is_touch_device() {
	return 'ontouchstart' in window // works on most browsers 
		|| 'onmsgesturechange' in window; // works on ie10
};

$(function() {

	/** Debug **/
	var debug = false;
	if (debug) $('body').addClass('debug');

	/** Touch **/
	var touch = false;
	if (is_touch_device()) {
		touch = true;
		$('body').addClass('touch');
	}

	/** Canvas **/
	var canvasDown = $('#down canvas')[0].getContext('2d');
	canvasDown.beginPath();
	canvasDown.moveTo(10, 10);
	canvasDown.lineTo(50, 30);
	canvasDown.lineTo(90, 10);
	canvasDown.lineWidth = 3;
	canvasDown.strokeStyle = 'rgba(255,255,255,0.9)';
	canvasDown.stroke();

	var canvasUp = $('#up canvas')[0].getContext('2d');
	canvasUp.beginPath();
	canvasUp.moveTo(5, 25);
	canvasUp.lineTo(20, 15);
	canvasUp.lineTo(35, 25);
	canvasUp.lineWidth = 2;
	canvasUp.strokeStyle = 'rgba(255,255,255,0.9)';
	canvasUp.stroke();

	var canvasPrev = $('#scrollPrev canvas')[0].getContext('2d');
	canvasPrev.beginPath();
	canvasPrev.moveTo(20, 15);
	canvasPrev.lineTo(12, 25);
	canvasPrev.lineTo(20, 35);
	canvasPrev.lineWidth = 2;
	canvasPrev.strokeStyle = 'rgba(255,255,255,0.9)';
	canvasPrev.stroke();

	var canvasNext = $('#scrollNext canvas')[0].getContext('2d');
	canvasNext.beginPath();
	canvasNext.moveTo(15, 15);
	canvasNext.lineTo(23, 25);
	canvasNext.lineTo(15, 35);
	canvasNext.lineWidth = 2;
	canvasNext.strokeStyle = 'rgba(255,255,255,0.9)';
	canvasNext.stroke();

	/** Map Close Popup Hack**/
	var openPopupId = 0;
	L.Map = L.Map.extend({
		openPopup: function(popup) {
			//console.log(openPopupId + ", " + popup._leaflet_id);
			//if (openPopupId != popup._leaflet_id) {
				this.closePopup();
				//closePopup();
				this._popup = popup;
				//openPopupId = popup._leaflet_id;
				return this.addLayer(popup).fire('popupopen', {
					popup: this._popup
				});
			//}
		}
	});

	/** Map **/
	map = L.map('map', {
		legendControl: false,
		infoControl: false,
		//fullscreenControl: true,
		scrollWheelZoom: false,
		zoomControl: false,
		attributionControl: false,
		center: [50, -40],
		zoom: 3,
		minZoom: 2,
		maxZoom: 7,
		worldCopyJump: true,
		//closePopupOnClick: false,
		//tap: false,
		//dragging: false,
	});
	map.addLayer(new L.TileLayer('images/maptiles/{z}/{x}/{y}.png'));
	map.setMaxBounds([[-180,-180], [180,180]]);
	L.control.zoom({
		position: 'topright'
	}).addTo(map);

	/** Country Layer **/
	/* function addCountry(code) {
		countryLayer[code] = L.mapbox.featureLayer().addTo(map);
		countryLayer[code].on('layeradd', function(e) {
			e.layer.setStyle({
				className: 'country',
				stroke: false,
				fill: true,
				fillOpacity: 0
			});
		});
		countryLayer[code].setGeoJSON(country[code]);
		countryLayer[code].eachLayer(function(layer) {
			var content = '<h2>' + layer.feature.properties.name + '<\/h2>';
			layer.bindPopup(content);
		});
	}
	var countries = ['US','CN','DE','GB','JP','KR','FR','ES','CA','AU','NL','NZ','IE','IT','SE','RU','IN','AT','ZA','CH'];
	$.each(countries, function(index, country) {
		addCountry(country);
	}); */

	/** Collision Detection **/
	var oms = new OverlappingMarkerSpiderfier(map);

	/** Brand Layer **/
	dataLayer = L.mapbox.featureLayer().addTo(map);

	/** Marker **/
	dataLayer.on('layeradd', function(e) {
		e.layer.setIcon(
			L.divIcon({
				className: 'marker ' + e.layer.feature.properties.Category.replace(/ /g,'').toLowerCase(),
				iconSize: [e.layer.feature.properties.Pin,e.layer.feature.properties.Pin],
				html: '<div class="markerInner">&nbsp;</div>'
			})
		);
		oms.addMarker(e.layer);
	});
	dataLayer.setGeoJSON(data);

	/** Tooltips **/
	var chartData = [];
	var chartOptions = [];
	function tooltips() {
		unBindLayerEvents();
		dataLayer.eachLayer(function(layer) {
			var brand = layer.feature.properties.Brand.replace(/ /g,'').toLowerCase();
			var category = layer.feature.properties.Category.replace(/ /g,'').toLowerCase();
			var content =
				'<div class="img"><img src="images/logos/' + brand + '.svg" alt="' + layer.feature.properties.Brand + '" /></div>' +
				'<div class="info">' +
					'<div class="left">' +
						'<div class="headline">Brand Value</div>' +
						'<div class="value">' +
							'<div class="currency">$</div>' +
							'<div class="number">' +
								'<div>' + layer.feature.properties.Value2018.toString().replace(/(\d)(?=(\d\d\d)+(?!\d))/g, "$1,") + '</div>' +
								'<div class="unit">Million</div>' +
							'</div>' +
						'</div>' +
					'</div>' +
					'<div class="right">' +
						'<div class="headline">Rank \'18</div>' +
						'<div class="rank">' + layer.feature.properties.Rank2018 + '.</div>' +
					'</div>' +
				'</div>' +
				'<div class="chart"><canvas id="chart-' + layer.feature.properties.Rank2018 + '"></canvas></div>';
			layer.bindPopup(content, {
				closeButton: false,
				className: category
			});
			var color = 'rgba(33,33,33,.9)';
			if (colors[category]) color = 'rgba(' + colors[category+'RGB'] + ',.9)';
			var textColor = '#FFF';
			if (category == 'luxury') textColor = '#000';
			chartOptions[layer.feature.properties.Rank2018] = {
				inGraphDataFontColor: textColor
			};
			chartData[layer.feature.properties.Rank2018] = {
				labels: ['2010','2011','2012','2013','2014','2015','2016','2017','2018'],
				datasets: [{
					fillColor: color,
					strokeColor: 'rgba(0,0,0,0)',
					data: [layer.feature.properties.Value2010,layer.feature.properties.Value2011,layer.feature.properties.Value2012,layer.feature.properties.Value2013,layer.feature.properties.Value2014,layer.feature.properties.Value2015,layer.feature.properties.Value2016,layer.feature.properties.Value2017,layer.feature.properties.Value2018]
				}]
			};
		});
		
		bindLayerEvents();
		bindMarkerEvents();
	}

	function bindLayerEvents() {
		dataLayer.on('mouseover click', function(e) {
			var $icon = $(e.layer._icon);
			if ($icon.data('transition-status') === 'finished' || !$icon.hasClass('spiderfied')) {
				openPopup(e);
			}
		});
		dataLayer.on('mouseout', function(e) {
			//closePopup(e);
		});
	}	
	function unBindLayerEvents() {
		dataLayer.off('mouseover click');
	}

	function bindMarkerEvents() {
		$('.marker').on('transitionend webkitTransitionEnd oTransitionEnd MSTransitionEnd', function() {
			if ($(this).data('transition-status') === 'deleted') $(this).data('transition-status', null);
			else $(this).data('transition-status', 'finished');
			return false;
		});
	}

	/** Popup **/
	function openPopup(e) {
		var popup = e.layer.openPopup();
		var $popupPane = $('.leaflet-popup-pane');		
		var $popup = $('.leaflet-popup');
		var $popupHtml = '<div class="popup-container">' + $popup.html() + '</div>';
		$popup.html($popupHtml);
		generateChart(e.layer.feature.properties.Rank2018);
	}
	function closePopup(e) {
		/*$('.leaflet-popup').addClass('popupclose');
		window.setTimeout(function(){
			$('.leaflet-popup').removeClass('popupclose');
		},200);*/
	}

	/** Tooltip Charts **/
	var ctx = [];
	var chart = [];
	function generateChart(chartnr) {
		ctx[chartnr] = $('#chart-'+chartnr).get(0).getContext("2d");
		chart[chartnr] = new Chart(ctx[chartnr]);
		//$('#chart-'+chartnr)[0].maintainAspectRatio = false;
		//$('#chart-'+chartnr)[0].width = 250;
		//$('#chart-'+chartnr)[0].height = 200;
		chart[chartnr].HorizontalBar(chartData[chartnr],chartOptions[chartnr]);
	}

	/** Filter **/
	function filterDataLayer(filter) {
		$('#map').removeClass('spiderfied');
		$('.marker').removeClass('spiderfied');
		if (!fadeout) {
			fadeout = true;
			$('.marker').addClass('fadeout');
			window.setTimeout(setFilter,200);
		}
		function setFilter () {
			fadeout = false;
			dataLayer.setFilter(function(feature) {
				if (!filter || filter == 'false') return false;
				if (filter == 'All') return true;
				return feature.properties.Category === filter;
			});
			tooltips();
		}
	}
	filterDataLayer();

	/** Scrolling **/
	var box = $('#content section'), boxDone = [false,false,false,false,false,false,false,false,false,false,false], navElem = $('#nav li'), filterDone = false;
	var scroll = skrollr.init({
		forceHeight: true,
		smoothScrolling: true,
		render: function() {
			box.each(function(index) {
				if ($(this).hasClass('skrollable-between')) {
					var navItem = $('#nav li:eq(' + index + ')');
					if (target && target != navItem.text()) return;
					if (target && target == navItem.text()) target = false;
					filterDone = false;
					if (!boxDone[index]) {
						$('svg.leaflet-zoom-animated path').css('stroke','transparent');
						navElem.removeClass('active');
						navItem.addClass('active');
						filterDataLayer(navItem.text());
						setHash($(this).attr('id'));
						boxDone[index] = true;
					}
				} else if (boxDone[index]) {
					boxDone[index] = false;
				}
			});
			var activeFilter = $('section.skrollable-between');
			if (activeFilter.length == 0 && !filterDone) {
				navElem.removeClass('active');
				filterDataLayer();
				filterDone = true;
			}
		},
		mobileCheck: function() {
			return false;
		}
	});

	/** Hash Navigation **/
	var pageReady = false;
	var target;
	skrollr.menu.init(scroll, {
    animate: true,
    easing: 'linear',
    scale: 1,
    duration: function(currentTop, targetTop) {
    	return 500;
    	/* if (currentTop > targetTop) return (currentTop-targetTop)/4;
    	else return (targetTop-currentTop)/4; */
    },
    handleLink: function(link) {
			if (pageReady) {
				target = $(link).text();
				return $(link).data('scroll');
			}
			return false;
    }
	});
	Pace.on('done', function() {
		pageReady = true;
		if (hash = (window.location.hash).substr(2)) {
			target = $('#'+hash).text();
			scroll.animateTo($('#'+hash).data('scroll'));
		}
	});
	function setHash (hash) {
		history.pushState(null, null, '#/'+hash);
	}

	/** Navigation Scroller **/
	var navScroller = $('#navBar');
	var navScrollerElem = $('#navBar ul');
	var scrollPrev = $('#scrollPrev');
	var scrollNext = $('#scrollNext');
	var posLeft, posRight, posLeftOk = false, posRightOk = false;
	function checkNavScroller() {
		posLeft = navScrollerElem.position().left;
		if (posLeft < 0 && !posLeftOk) {
			scrollPrev.addClass('active');
			registerNavScroller(scrollPrev, 'prev');
			posLeftOk = true;
		} else if (posLeft >= 0) {
			scrollPrev.removeClass('active');
			posLeftOk = false;
		}
		posRight = (posLeft - navScroller.width()) + navScrollerElem.width();
		if (posRight > 0 && !posRightOk) {
			scrollNext.addClass('active');
			registerNavScroller(scrollNext, 'next');
			posRightOk = true;
		} else if (posRight <= 0) {
			scrollNext.removeClass('active');
			posRightOk = false;
		}
	}
	navScroller.on('scroll', function() {
		checkNavScroller();
	});
	$(window).on('resize', function(){
		checkNavScroller();
	});
	checkNavScroller();
	function clickNavScroller(direction) {
		if (direction == 'prev') {
			navScroller.animate({
				scrollLeft: -(posLeft+150)
			}, 300);
		}
		else if (direction == 'next') {
			navScroller.animate({
				scrollLeft: -(posLeft-150)
			}, 300);
		}
	}
	function registerNavScroller(elem, direction) {
		elem.on('click', function() {
			clickNavScroller(direction);
		});
	}

});
